src/image_ops.rs

```diff
@@ -1582,6 +1584,134 @@ Images are first converted to grayscale. Thresholds should be `>=0` and `< 256`.
         Example::new(1, 1, "thresh 120")
 );
 
+//-----------------------------------------------------------------------------
+// Threshold
+//-----------------------------------------------------------------------------
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+struct Tile(u32, u32);
+
+impl ImageOp for Tile {
+    fn apply(&self, stack: &mut ImageStack) {
+        let mut image = stack.pop();
+
+        let tiles = tile(&mut image, self.0, self.1);
+
+        for t in tiles.into_iter().rev() {
+            stack.push(t)
+        }
+    }
+
+    // takes 1, returns n
+    fn signature(&self) -> Option<(usize, usize)> {
+        None
+    }
+}
+
+fn tile(mut image: &mut DynamicImage, width: u32, height: u32) -> Vec<DynamicImage> {
+    let max_elem = (image.width() / width + 1) * (image.height() / height + 1);
+    let mut out: Vec<DynamicImage> = Vec::with_capacity(max_elem as usize);
+
+    let background = match image {
+        ImageLuma8(_) => ImageLuma8(ImageBuffer::from_pixel(width, height, Luma([0]))),
+        ImageLumaA8(_) => ImageLumaA8(ImageBuffer::from_pixel(width, height, LumaA([0, 0]))),
+        ImageRgb8(_) => ImageRgb8(ImageBuffer::from_pixel(width, height, Rgb([0, 0, 0]))),
+        ImageRgba8(_) => ImageRgba8(ImageBuffer::from_pixel(width, height, Rgba([0, 0, 0, 0]))),
+        ImageBgr8(_) => ImageBgr8(ImageBuffer::from_pixel(width, height, Bgr([0, 0, 0]))),
+        ImageBgra8(_) => ImageBgra8(ImageBuffer::from_pixel(width, height, Bgra([0, 0, 0, 0]))),
+    };
+
+    for ymin in (0u32..image.height()).step_by(height as usize) {
+        let ymax = ymin + height;
+
+        for xmin in (0u32..image.width()).step_by(width as usize) {
+            let xmax = xmin + width;
+
+            let crop_spec = Crop {
+                left: xmin,
+                top: ymin,
+                width,
+                height,
+            };
+            let over = dynamic_map!(&mut image, |i| crop(i, &crop_spec));
+
+            if xmax > image.width() || ymax > image.height() {
+                let under = background.clone();
+
+                // todo: this is copied from Overlay, try to factor it out
```

Comment:
Let's bin the padding, and then this can just go. (As you've spotted, working with `DynamicImage`s is currently a bit awkward. I'm not sure how much cleaner we can make this.)